src/lib/services/embed-url-parser-service/embed-url-parser-service.ts

```diff
@@ -21,6 +21,17 @@ export class EmbedUrlParserService {
     return youtubeVideoID ? `https://www.youtube.com/embed/${youtubeVideoID}` : "";
   }
 
+  static mousaiParser(url): string | boolean {
+    const regExp = /^.*mousai\.stream\/((album|playlist|track)\/[a-z0-9-_]+)(?=\/embed|$)/;
```

Comment:
```suggestion
    const regExp = /^.*mousai\.stream\/((album|playlist|track)\/[0-9]+(\/[a-z0-9-_]+){1,2}?)(?=(\/embed|$))/;
```

this regex will more directly match your URL format